keg/generator.py

```diff
@@ -19,69 +19,92 @@
 from jinja2 import Environment, FileSystemLoader
 import os
 
+from keg.image_definition import KegImageDefinition
+from keg.kiwi_description import KiwiDescription
 from keg import (
-    image_definition,
-    kiwi_description,
-    template_functions,
-    utils
+    template_functions, utils
 )
-from keg.exceptions import KegError, KegKiwiValidationError
+from keg.exceptions import KegError
 
 log = logging.getLogger('keg')
 
 
-def create_image_description(
-    image_source, recipes_root, data_roots,
-    dest_dir, force=False
-):
+class KegGenerator:
     """
-    Create a KIWI image description in dest_dir from image_source.
+    Class for creating a KIWI image description from Keg Templates
 
-    Expects a keg-recipes structure at recipes_root and image_source
-    referring to a directory under images in recipes_root.
+    :param object image_definition: Instance of KegImageDefinition
+    :param str dest_dir: Destination directory
     """
-    # Sanity check image source location
-    if not os.path.exists(os.path.join(recipes_root, 'images', image_source)):
-        errmsg = 'Source directory for {} does not exist.'.format(image_source)
-        if force:
-            log.warning(errmsg)
-        else:
-            raise KegError(errmsg)
+    def __init__(
+        self, image_definition: KegImageDefinition, dest_dir: str
+    ):
+        self.image_definition = image_definition
+        self.description_schemas = os.path.join(
+            image_definition.recipes_root, 'schemas'
+        )
+        self.dest_dir = dest_dir
```

Comment:
IMHO that dest_dir is in fact a directory should be checked 

if no os.path.isdir(dest_dir):
  COMPLAIN

Reply:
yes fully agree and done